gloo-net: Cannot compile with `http` feature but not `json` feature

### Describe the Bug

The `.json` helper on `Request` (added in #204) is not conditional on the `json` feature. The same goes for the `serde::Serialize` import at the top of the file.

### Steps to Reproduce

1. Add `gloo-net = { version = "0.2.1", default-features = false, features = [ "http" ] }` to Cargo.toml
2. Run `cargo build`

### Expected Behavior

The `.json` helper is not available because the `json` feature is not enabled and the crate compiles.

### Actual Behavior

The crate fails to compile because of the unresolved import:

```
error[E0432]: unresolved import `serde`
  --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/gloo-net-0.2.1/src/http/mod.rs:21:5
   |
21 | use serde::Serialize;
   |     ^^^^^ use of undeclared crate or module `serde`

error[E0433]: failed to resolve: use of undeclared crate or module `serde_json`
   --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/gloo-net-0.2.1/src/http/mod.rs:176:20
    |
176 |         let json = serde_json::to_string(value)?;
    |                    ^^^^^^^^^^ use of undeclared crate or module `serde_json`
```

### Additional Context

--- a/.github/workflows/lint-fmt.yml
+++ b/.github/workflows/lint-fmt.yml
@@ -61,4 +61,11 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --all-targets -- -D warnings
+          args: --all-targets --all-features -- -D warnings
+
+      - name: Run clippy for gloo-net
+        working-directory: crates/net
+        run: |
+          cargo clippy --features "http"
+          cargo clippy --features "http,json"
+          cargo clippy --features "websocket"

--- a/crates/net/src/http/mod.rs
+++ b/crates/net/src/http/mod.rs
@@ -177,7 +177,7 @@ impl Request {
         let json = serde_json::to_string(value)?;
         Result::Ok(self.header("Content-Type", "application/json").body(json))
     }
-    
+
     /// The request method, e.g., GET, POST.
     pub fn method(mut self, method: Method) -> Self {
         self.options.method(&method.to_string());

--- a/crates/net/src/http/mod.rs
+++ b/crates/net/src/http/mod.rs
@@ -18,7 +18,6 @@ mod query;
 
 use crate::{js_to_error, Error};
 use js_sys::{ArrayBuffer, Uint8Array};
-use serde::Serialize;
 use std::fmt;
 use wasm_bindgen::prelude::*;
 use wasm_bindgen::JsCast;
@@ -167,15 +166,18 @@ impl Request {
         self.options.integrity(integrity);
         self
     }
+
     /// A convenience method to set JSON as request body
     ///
     /// # Note
     ///
     /// This method also sets the `Content-Type` header to `application/json`
-    pub fn json<T: Serialize + ?Sized>(self, value: &T) -> Result<Self, Error> {
+    #[cfg(feature = "json")]
+    pub fn json<T: serde::Serialize + ?Sized>(self, value: &T) -> Result<Self, Error> {
         let json = serde_json::to_string(value)?;
         Result::Ok(self.header("Content-Type", "application/json").body(json))
     }
+    
     /// The request method, e.g., GET, POST.
     pub fn method(mut self, method: Method) -> Self {
         self.options.method(&method.to_string());

